lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -543,23 +556,87 @@ class LearnerProgramEnrollmentsView(DeveloperErrorViewMixin, APIView):
     )
     permission_classes = (IsAuthenticated,)
 
+    DEFAULT_PROGRAM_TYPE = 'masters'
+
     def get(self, request):
         """
         How to respond to a GET request to this endpoint
         """
-        program_enrollments = ProgramEnrollment.objects.filter(
-            user=request.user,
-            status__in=('enrolled', 'pending')
-        )
 
-        uuids = [enrollment.program_uuid for enrollment in program_enrollments]
+        request_user = request.user
+
+        catalog_data_of_programs = []
+        program_type_filter = request.GET.get('type', self.DEFAULT_PROGRAM_TYPE)
+
+        if request_user.is_staff:
+            all_programs = get_programs(request.site)
```

Comment:
As we discussed in person, I'm going to help create an index in the cache of program UUIDs by program type.